### PR TITLE
C#: Port the existing compiler-tracing.spec files to Lua.

### PR DESCRIPTION
**Pull request by criemen**
Right now, this is an unmodified port, without the additional functionality for `dotnet build` that we envision.

This means that we can switch back and forth between Lua and legacy tracing (ideally) without user-impacting differences.

### CODE REVIEW

**Review by cklin — APPROVED** [review]
I think the titles of the commit and the PR should both mention csharp specifically to reduce ambiguity.  Other than that 👍

**Comment by criemen** [comment]
@cklin good point, thanks. Fixed now.

**Review by cklin — APPROVED** [review]
👍 but I don't have write access so you will have to get approval from someone else too.

File: csharp/tools/tracing-config.lua [hunk 1]
```diff
@@ -0,0 +1,55 @@
+function RegisterExtractorPack()
+    local extractor = GetPlatformToolsDirectory() ..
+                          'Semmle.Extraction.CSharp.Driver'
+    if OperatingSystem == 'windows' then
+        extractor = GetPlatformToolsDirectory() ..
+                        'Semmle.Extraction.CSharp.Driver.exe'
```
Comment (hvitved) [inline csharp/tools/tracing-config.lua:6]: I don't know lua, but could you simply append `.exe` to `extractor` in this case?

**Review by hvitved — COMMENTED** [review]
Looks plausible to me.

File: csharp/tools/tracing-config.lua (hunk 1 above)
Comment (criemen) [reply csharp/tools/tracing-config.lua]: Good point, yes that's possible

**Review by hvitved — APPROVED** [review]

**Pull request merged**